### DISCUSSION TITLE
How to configure xref to find C symbol definitions?

### DISCUSSION THREAD

**Discussion by Luctins** [syl20bnr/spacemacs#15149] [Q&A]
I've tried to use `xref-find-definitions` bound to `M-.` in `c-mode`, but even definitions in the same file are not found. Do I need to configure it in some way? 

Into looking into this I've found many guides explaining how to set up rtags, can it be used as a backend for xref, like company mode's backends?

Thanks,

**Comment by lebensterben** [comment]
Best solution is to use one of the lsp.

Alternatively rtag works well with autocomplete layer.

**Reply by Luctins** [reply]
How do I configure rtags as a backend? I couldn't find the correct symbol to add to `company-backends`

**Reply by lebensterben** [reply]
https://github.com/syl20bnr/spacemacs/blob/4ebd512d140f5743ec731ab5147330ccf0e626c4/layers/%2Blang/c-c%2B%2B/packages.el#L48

No configuration needed.

**Reply by Luctins** [reply]
oh, ok, very simple indeed.

Also related, does xref uses it as a backend? for me `M-.` (`xref-find-definitios`) always returns `user-error: No definitions found for: NAME` where name is the symbol I am looking for.

**Reply by lebensterben** [reply]
there's https://melpa.org/#/rtags-xref

maybe this is not used in Spacemacs.
